nntrainer/include/conv2d_layer.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0-only
+ *
+ * @file	conv2d_layer.h
+ * @date	01 June 2020
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Jijoong Moon <jijoong.moon@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is Convolution Layer Class for Neural Network
+ *
+ */
+
+#ifndef __CONV2D_LAYER_H_
+#define __CONV2D_LAYER_H_
+#ifdef __cplusplus
+
+#include <fstream>
+#include <iostream>
+#include <layer.h>
+#include <optimizer.h>
+#include <tensor.h>
+#include <vector>
+
+#define CONV2D_DIM 2
+
+namespace nntrainer {
+
+/**
+ * @class   Convolution 2D Layer
+ * @brief   Convolution 2D Layer
+ */
+class Conv2DLayer : public Layer {
+public:
+  /**
+   * @brief     Constructor of Conv 2D Layer
+   */
+  Conv2DLayer() { setType(LAYER_CONV2D); };
+
+  /**
+   * @brief     Destructor of Conv 2D Layer
+   */
+  ~Conv2DLayer(){};
+  
+  /**
+   * @brief     initialize layer
+   * @param[in] last last layer
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  int initialize(bool last);
+
+  /**
+   * @brief     Read Weight & Bias Data from file
+   * @param[in] file input stream file
+   */
+  void read(std::ifstream &file);
+
+  /**
+   * @brief     Save Weight & Bias Data to file
+   * @param[in] file output stream file
+   */
+  void save(std::ofstream &file);
+
+  /**
+   * @brief     forward propagation with input
+   * @param[in] in Input Tensor from upper layer
+   * @retval    Activation(W x input + B)
+   */
+  Tensor forwarding(Tensor in, int &status);
+
+  /**
+   * @brief     back propagation
+   *            Calculate dJdB & dJdW & Update W & B
+   * @param[in] input Input Tensor from lower layer
+   * @param[in] iteration Number of Epoch for ADAM
+   * @retval    dJdB x W Tensor
+   */
+  Tensor backwarding(Tensor in, int iteration);
+
+  /**
+   * @brief     copy layer
+   * @param[in] l layer to copy
+   */
+  void copy(std::shared_ptr<Layer> l);
+
+  /**
+   * @brief     set Property of layer
+   * @param[in] values values of property
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  int setProperty(std::vector<std::string> values);
+
+  /**
+   * @brief     Property Enumeration
+   *            0. input shape : string
+   *            1. bias zero : bool
+   *            4. activation : bool
+   *            6. weight_decay : string (type)
+   *            7. weight_decay_lambda : float
+   *            9. filter_size : int
+   *            10. kernel_size : ( n , m )
+   *            11. stride : ( n, m )
+   *            12, padding : valid | same
```

Comment:
missing `weight_ini` and number after 9 needs correction.

Reply:
fixed. thanks.